File: src/main/java/ru/uvarov/flashcards/model/Answer.java
package ru.uvarov.flashcards.model;


import java.util.Objects;

public record Answer(
    String word,
    String translate,
    boolean isRight
) {
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Answer answer = (Answer) o;
        return word.equals(answer.word) && translate.equals(answer.translate);
    }

    @Override
    public int hashCode() {
        return Objects.hash(word, translate);
    }
}

File: src/main/kotlin/ru/uvarov/flashcards/model/Answer.kt
package ru.uvarov.flashcards.model

import java.util.Objects

data class Answer(
    val word: String,
    val translate: String,
    val correct: Boolean,
) {
    // equals/hashCode игнорируют correct: дедупликация в QuizService.fillIfNotEnough
    // должна считать «правильный» и «случайно совпавший» одним и тем же ответом.
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Answer) return false
        return word == other.word && translate == other.translate
    }

    override fun hashCode(): Int = Objects.hash(word, translate)
}
